src/server/routing/router-public.ts

```diff
@@ -164,6 +166,42 @@ router.get(
   }
 );
 
+// /**
+//  * Zaak status endpoint redirects to zaakstatus if authenticated, else redirect to login
+//  */
+router.get(BffEndpoints.ZAAK_STATUS, zaakStatusHandler);
+
+export async function zaakStatusHandler(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const authProfileAndToken = getAuth(req);
+    const params = queryParams<{
+      id: string;
+      thema: string;
+      'auth-type': string;
+    }>(req);
+
+    const redirectUrl = getReturnToUrlZaakStatus(params);
+
+    if (authProfileAndToken) {
+      return res.redirect(`${redirectUrl}`);
+    }
+
+    const authType =
+      params['auth-type'] === 'eherkenning' ? 'EHERKENNING' : 'DIGID';
+    const loginRoute = authRoutes[`AUTH_LOGIN_${authType}`];
+
+    const loginRouteWithReturnTo = `${loginRoute}?returnTo=${AppRoutes.ZAAK_STATUS}`;
```

Comment:
Je geeft de query params die nodig zijn voor de constructie van de correcte frontend url (in getReturnUrl() bij de login endpoints) niet mee.